pkg/plugins/backendplugin/coreplugin/core_plugin.go

```diff
@@ -14,18 +14,24 @@ import (
 type corePlugin struct {
 	pluginID string
 	logger   log.Logger
+	opts     backend.ServeOpts
 	backend.CheckHealthHandler
 	backend.CallResourceHandler
 	backend.QueryDataHandler
 	backend.StreamHandler
 }
 
+var _ plugins.CorePluginWithServerOpts = (*corePlugin)(nil)
```

Comment:
alternatively we could make `corePlugin` public -- then we can check if casting works